mock-identity-certify-plugin/src/main/java/io.mosip.certify.mock.integration/service/MockVCIssuancePlugin.java

```diff
@@ -0,0 +1,245 @@
+import io.mosip.kernel.core.util.CryptoUtil;
+import io.mosip.kernel.keymanagerservice.constant.KeymanagerConstant;
+import io.mosip.kernel.keymanagerservice.entity.KeyAlias;
+import io.mosip.kernel.keymanagerservice.helper.KeymanagerDBHelper;
+import io.mosip.kernel.signature.dto.JWTSignatureRequestDto;
+import io.mosip.kernel.signature.dto.JWTSignatureResponseDto;
+import io.mosip.kernel.signature.service.SignatureService;
+import lombok.extern.slf4j.Slf4j;
+
+@ConditionalOnProperty(value = "mosip.certify.integration.vci-plugin", havingValue = "MockVCIssuancePlugin")
+@Component
+@Slf4j
+public class MockVCIssuancePlugin implements VCIssuancePlugin {
+
+	private static final String AES_CIPHER_FAILED = "aes_cipher_failed";
+	private static final String NO_UNIQUE_ALIAS = "no_unique_alias";
+	private static final String USERINFO_CACHE = "userinfo";
+
+	@Autowired
+	private SignatureService signatureService;
+
+	@Autowired
+	private ParsedAccessToken parsedAccessToken;
+
+	@Autowired
+	private CacheManager cacheManager;
+
+	@Autowired
+	private KeyStore keyStore;
+
+	@Autowired
+	private KeymanagerDBHelper dbHelper;
+
+	private ConfigurableDocumentLoader confDocumentLoader = null;
+
+	@Value("${mosip.certify.mock.vciplugin.verification-method}")
+	private String verificationMethod;
+
+	@Value("${mosip.certify.mock.authenticator.get-identity-url}")
+	private String getIdentityUrl;
+
+	@Value("${mosip.certify.cache.security.secretkey.reference-id}")
+	private String cacheSecretKeyRefId;
+
+	@Value("${mosip.certify.cache.security.algorithm-name}")
+	private String aesECBTransformation;
+
+	@Value("${mosip.certify.cache.secure.individual-id}")
+	private boolean secureIndividualId;
+
+	@Value("${mosip.certify.cache.store.individual-id}")
+	private boolean storeIndividualId;
+
+	@Value("#{${mosip.certify.mock.vciplugin.vc-credential-contexts:{'https://www.w3.org/2018/credentials/v1','https://schema.org/'}}}")
+	private List<String> vcCredentialContexts;
+
+	public static final String UTC_DATETIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+
+	public static final String CERTIFY_SERVICE_APP_ID = "CERTIFY_SERVICE";
+
+	@Override
+	public VCResult<JsonLDObject> getVerifiableCredentialWithLinkedDataProof(VCRequestDto vcRequestDto, String holderId,
+																			 Map<String, Object> identityDetails) throws VCIExchangeException {
+		JsonLDObject vcJsonLdObject = null;
+		try {
+			VCResult<JsonLDObject> vcResult = new VCResult<>();
+			vcJsonLdObject = buildDummyJsonLDWithLDProof(holderId);
+			vcResult.setCredential(vcJsonLdObject);
+			vcResult.setFormat("ldp_vc");
+			return vcResult;
+		} catch (Exception e) {
+			log.error("Failed to build mock VC", e);
+		}
+		throw new VCIExchangeException();
+	}
+
+	private JsonLDObject buildDummyJsonLDWithLDProof(String holderId)
+			throws IOException, GeneralSecurityException, JsonLDException, URISyntaxException {
+		OIDCTransaction transaction = getUserInfoTransaction(parsedAccessToken.getAccessTokenHash());
```

Comment:
identity details does not have the `accessTokenHash` value

Reply:
Can we understand why is this the case ? Can we also understand how the bean approch will work ?

Reply:
identity details has the claims from decoded token, parsed access token has the `accessTokenHash` generated when validation token in the [AccessTokenValidationFilter](https://github.com/mosip/inji-certify/blob/3a8093589fbcaa6049c191941f01723e88f5d927/certify-service/src/main/java/io/mosip/certify/filter/AccessTokenValidationFilter.java#L92), since the bean is available in inji-certify plugin will be able to access it

Reply:
How till this work in a high loaded env ?